### ISSUE TITLE
VSCode always executes the "folderOpen" task after every single task

### ISSUE THREAD

**Issue by CrendKing** [microsoft/vscode#160013]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.71.0
- OS Version: Windows 11 22200.856

Steps to Reproduce:

0. The following is tested with completely clean VSCode.
1. Create a directory and then create .vscode directory under.
2. Create settings.json with content
```
{
    "task.allowAutomaticTasks": "on"
}
```
3. Create tasks.json with content
```
{
    "version": "2.0.0",
    "tasks": [
        {
            "label": "manual",
            "type": "shell",
            "command": "echo Hello",
            "problemMatcher": [],
        },
        {
            "label": "auto",
            "type": "shell",
            "command": "echo World",
            "runOptions": {
                "runOn": "folderOpen"
            }
        }
    ]
}
```
4. Close VSCode. Reopen the root directory with VSCode. Observe that the "auto" task is executed, as expected (because it's `folderOpen`).
5. Run task "manual" manually. Observe that the "auto" task is executed at the end of every time. This behavior only happens post VSCode 1.70.

I thought "folderOpen" means run the task when the folder is open, not when another task is executed?

**Comment by meganrogge** [comment]
I believe this is already fixed, can you see if you can reproduce in the [Insiders build](https://code.visualstudio.com/insiders)?
/label info-needed

**Comment by jdtcd** [comment]
Same issue reproduced with

VSCode Version 1.72.0-insider (Universal)
OS: Darwin arm64 21.6.0

**Comment by meganrogge** [comment]
reproduces in insider's but not in OSS

**Comment by CrendKing** [comment]
Just tried
```
Version: 1.72.0-insider
Commit: 0f2de31e4856886cdbf32f85488bc40f480fb38f
Date: 2022-09-07T05:17:13.628Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.22000
Sandboxed: Yes
```
and still repro:
```
Executing task: echo World 

Hello
 *  Terminal will be reused by tasks, press any key to close it. 

 
World
 *  Terminal will be reused by tasks, press any key to close it. 
```

**Comment by meganrogge** [comment]
yep as mentioned above I can reproduce

**Comment by meganrogge** [comment]
as you see here, I thought it was fixed but it hasn't been yet https://github.com/microsoft/vscode/issues/160013#issuecomment-1239973938

**Comment by meganrogge** [comment]
to verify:

to verify, open any trusted workspace and configure a task with `runOptions: { runOn: openFolder}`.

run a task and verify that the automatic task that you set doesn't get run

**Comment by CrendKing** [comment]
I don't know exactly which PR fixes this, but as of 5cecbf4746a7b95dedaf9fc4534b9786b9a30ace I can verify it's no longer an issue.
